### PR TITLE
New notes widget

### PR DESCRIPTION
**Pull request by shachar058**
Created another type of notes widget, can choose between them in Globals.

New type contains:
- ajax button to update message status to complete
- added 'To' column
- 'Content' contains only content itself without date or users

(below added by bradymiller)
Up For Grabs demo is here:
http://www.open-emr.org/wiki/index.php/Development_Demo#Mu_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by bradymiller** [comment]
To make it easy for other users to test this, can test on the following Up For Grabs demo:
http://www.open-emr.org/wiki/index.php/Development_Demo#Mu_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
hi @shachar058 ,
This is a nice feature and seems like it would make more sense to just use your feature (ie. no reason for the global and/or supporting the prior mechanism). Thoughts?
-brady

**Comment by shachar058** [comment]
@bradymiller 
We could do that, but it's important for us that the date does not contain seconds, and that the content does not contain the users and the date. If that's okay with you, then we'd be happy to just leave out the global.

**Comment by bradymiller** [comment]
hi @shachar058 ,
That is definitely ok with me and is more readable like that. What are your plans with the Status column (up to you, but your group did just add this)?
-brady

**Comment by shachar058** [comment]
Hi @bradymiller 
We added the status column in the full notes page, but we wanted to take it out of the widget, instead having the new button.

**Comment by bradymiller** [comment]
Sounds like a good plan.

**Comment by shachar058** [comment]
@bradymiller 
Hi,
I took out the global + old widget type like you asked.
thanks

**Comment by bradymiller** [comment]
Hi @shachar058 ,

The issue I noted is currently related to your groups PR on the multiple patients.

It seems like we are ignoring the current mechanism to send patient messages.
@sjpadgett , another question on your patient messaging mechanism. When a patient sends a message, does it show up in the `Sent Items` tab of the Messages section in patient summary widget
(for @sjpadgett , my concern here is that we have 3 different ways for patient messaging; I think having the onsite portal (and offsite portal) with different mechanism is acceptable as long as they both work together, but appears Matrix is headed towards another mechanism)

For @shachar058, let me wait for @sjpadgett' s input before I give recs (I will likely just rec. to use the offsite portal global to hide/show the  `Sent Items` tab in the widget, but let me get Jerry's thoughts first).

-brady

**Comment by sjpadgett** [comment]
image

I don't receive from pnotes but only send notifications to pnotes. New patient registered through onsite, A forwarded message from patient to portal dashboard i.e one user forwards to another user the patients secure msg from onsite and reminders if an invoice is sent to patient from posting. I don't look for any messages from pnote to portal, only ones sent from dashboard. This was model after the way V.A handles messaging. They screen secure messages and then route where appropriate.
Therefore portal messages concerning a patient will appear  in message center but not summary. A patient can't send a message directly to user via pnotes.
I think what Matrix wants to do is legit, although I never got the concept of how patient is to receive message via -patient- flag only that it use to work by allowing message to appear in summary. I don't think it is a new messaging system but just adding to existing.(correct me if i'm wrong). I'd just turn it on with one of the other globals they use.
Instead of billing maybe I ought to get a plan to unify messaging....

**Comment by bradymiller** [comment]
Sounds like the Inbox and Sent Items are then specific to the offsite portal form of messaging, where they used the -patient- flag to decide if the message was directed to(and I think also from) the patient. So, should probably show these tabs if the offsite portal globals flag is on.

**Comment by shachar058** [comment]
@bradymiller
Ok, I changed it so the tabs are shown if the offsite portal option is on.
Are we good to go? :)

File: interface/patient_file/summary/pnotes_fragment.php [hunk 1]
```diff
@@ -76,38 +84,30 @@
...
+                echo "<td valign='top' class='text'>".htmlspecialchars(date('Y-m-d H:i', strtotime($iter['date'])), ENT_NOQUOTES)."</td>\n";
                 echo "  <td valign='top' class='text'><b>";
                 echo generate_display_field(array('data_type'=>'1','list_id'=>'note_type'), $iter['title']);
                 echo "</b></td>\n";
 
                 echo "  <td valign='top' class='text'>$body</td>\n";
-                echo "  <td valign='top' class='text'>";
-                echo generate_display_field(array('data_type'=>'1','list_id'=>'message_status'), $iter['message_status']);
-                echo "</td>\n";
+                echo "<td valign='top' class='text'><button data-id='" . $iter['id'] . "' class='complete_btn'>" . htmlspecialchars('Completed', ENT_NOQUOTES) . "</button></td>\n";
```
Comment (bradymiller) [inline interface/patient_file/summary/pnotes_fragment.php]: translate Completed:
``````
xlt('Completed')

Comment (bradymiller) [reply interface/patient_file/summary/pnotes_fragment.php]: Also should escape $iter['id']
``````
attr($iter['id'])

File: interface/themes/style_light.css [hunk 2]
```diff
@@ -470,6 +470,10 @@ a:hover {
   margin: 5px;
 }
 
+#pnotes_ps_expand  div.tab table td {
+  padding-right: 10px;
+}
```
Comment (bradymiller) [inline interface/themes/style_light.css]: Is this something that needs to go into every theme?
(note there are 20+ of these now, and I'm happy to take on that glorious task, if needed. Just let me know.)

**Comment by bradymiller** [comment]
Pretty much. Just noted translations and escaping issue. Then it's ready to go. I can push that code block to all the other themes is you wish to avoid that joyous task :)

File: interface/patient_file/summary/pnotes_fragment.php (hunk 1 above)
Comment (shachar058) [reply interface/patient_file/summary/pnotes_fragment.php]: ok done, don't know how I missed that...

File: interface/themes/style_light.css (hunk 2 above)
Comment (shachar058) [reply interface/themes/style_light.css]: I think it should go in every theme, and would be **overjoyed** to hand this over to you :)

**Comment by bradymiller** [comment]
whoohoo, that was fun. bringing this into codebase. thanks for the very nice feature! -brady

**Pull request merged**